lib/flux/app-state.js

```diff
@@ -439,15 +439,27 @@ export const actionMap = new ActionMap({
       },
     },
 
+    syncNote: {
+      creator({ noteBucket, noteId }) {
+        console.log('sync the note');
+        return () => {
+          console.log('syncing');
+          noteBucket.touch(noteId);
+        };
```

Comment:
I'm pretty sure we don't need this thunk here and can just run the code and return

on that note when I was visualizing the change I had `syncNote()` inside of `app.jsx` - it doesn't change the app state in any way and it's independent from all the data flow in the app, therefore I think it may just complicate things by adding the additional step.

do you have strong feelings either way? keeping it here as a new action creator vs. leaving it in `app.jsx`

```js
syncNote = noteId => {
  const { noteBucket } = this.props;

  noteBucket.touch( noteId );
};

<AppLayout
	…
	syncNote={ this.syncNote }
	…
/>
```

Reply:
> do you have strong feelings either way?

I was naively copying the action for updating the note content so no strong opinions.

Reply:
if you have no strong opinion let's keep it in `app.jsx` since we don't have any real reason to add the indirection

if it impacted app state at all or didn't represent something that might be happening in the background anyway I'd feel more compelled to put it in an action creator, but `app-state` is already confusing enough

Reply:
Do you want all of the props to be called `syncNote` as well?

Reply:
to me that makes sense. it's not even arguable in the way that `onNoteChanged` is, where the author changed the note and the app is responding. in this case it's purely the app acting and doing so actively, so `onNoteSync` leads me to believe there is some other actor involved when actually there isn't

Reply:
For me when it comes to React components I tend to treat callback functions as a signal. The `NoteDetail` isn't trying to sync the note, but it is saying "whoever is giving me my props, it's time to sync the note".

In that regard, it is not doing the syncing itself but controlling the timing of it. The convention I end up with is: A React components props that are callbacks for future things that happen start with `on`, so `onMySignal`.

Wether the note is synced or not `NoteDetail` no longer worries about that.

Given that `AppLayout` is now doing the call to `noteBucket.touch()` I think having its function named `syncNote` makes the most sense. I'm still partial to component props having `onMySignal`. But no to the point that I'll fight very hard for it.

TL;DR I'll change it all to `syncNote={}`.